src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
 import { UrlBasedRequestPrepareOptions } from 'ms-rest';
+import { stringCompareFunction } from "../../helper/commonHelper";
 import { GitHubProvider } from "../../helper/gitHubHelper";
 import { SodiumLibHelper } from "../../helper/sodium/SodiumLibHelper";
+import { GitHubOrganization, GitHubRepo } from '../../model/models';
 import { RestClient } from "../restClient";
+const uuid = require('uuid/v4');
 
 const UserAgent = "deploy-to-azure-vscode";
 
 export class GithubClient {
 
     private patToken: string;
     private url: string;
+    private listOrgPromise: Promise<void | GitHubOrganization[]>;
+
 
     constructor(patToken: string, remoteUrl: string) {
         this.patToken = patToken;
         this.url = remoteUrl;
+        this.listOrgPromise = this.listOrganizations(this.patToken);
     }
 
-    public async createOrUpdateGithubSecret(secretName: string, body: string): Promise<void> {
-        let secretKeyObject: GitHubSecretKey = await this._getGitHubSecretKey();
-        let sodiumObj = new SodiumLibHelper(secretKeyObject.key);
-        let encryptedBytes: Uint8Array = sodiumObj.encrypt(body);
-        let encryptedBytesAsString: string = SodiumLibHelper.convertUint8ArrayToString(encryptedBytes);
-        let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
-        await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+    public createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+        let restClient =new RestClient();
+        let Url = "https://api.github.com/orgs/" + orgName + "/repos";
+        try{
+            return restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
+                url: Url,
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + this.patToken,
+                    "User-Agent": UserAgent
+                },
+                method: 'POST',
+                body: {
+                    "name": repoName,
+                    "description": "Repo created from VScode",
+                    "homepage": "https://github.com",
+                    "private": false,
+                    "has_issues": true,
+                    "has_projects": true,
+                    "has_wiki": true
+                },
+                deserializationMapper: null,
+                serializationMapper: null
+            })
+                .then((detail: GitHubRepo) => {
+                    return detail;
+                }).catch(error=>{
+                        return null;
```

Comment:
why do you want to suppress the error and return null instead?

Reply:
Because there are considerable chances of occurrence of an error (already existing repo of the same name),  therefore the error is suppressed here and void is returned. On receiving 'void', caller function will know that the repo name already exists and therefore will append organization name or uuid to make the repo name unique.

Reply:
Isn't the failure may also happen when user doesn't have permission to create repo in the current org?? So it is better to bump up the error to the user.

Reply:
That is correct. I think you should pass a variable `shouldThrow` as argument. If true, then do not suppress and throw. caller will set `false` first two times but will set `true` the last time.

Reply:
@bishal-pdMSFT  If repository cannot be created, then localGithubWorkflowConfigurer.ts is already throwing an error. Should I just change the error message to "GitHub repository cannot be created. You either don't have permission to create a repository inside the selected organization or a repository with the same name already exists."